### DISCUSSION TITLE
Unable to update VM attached IP to Primary IP via Rest API

### DISCUSSION THREAD

**Discussion by akballow** [netbox-community/netbox#6151] [Q&A]
### NetBox version

v2.10.4

### Python version

3.6

### Steps to Reproduce

1. Create VM Manually - TestVM
2. Create Interface Manually - TestVMnic
3. Attach an IP Manually with setting as primary- 1.1.1.1/24
4. Run API patch call to make the attached IP become the primary
curl -s "https://netbox/api/virtualization/virtual-machines/1/" \
  -X PATCH \
  -d '{"primary_ip": {"id": 1}}' \
  -H "Authorization: Token XXXX" \
  -H "Content-Type: application/json" | jq

Nothing happends

I tried variances like '{"primary_ip": {"id": 1}}' '{"primary_ip":  1}' and '{"primary_ip": "1"}'

Without any luck

### Expected Behavior

The non primary IP is not the Primary IP fore the VM.

I do not get this issue for non VM devices.

### Observed Behavior

Nothing Happens

**Comment by jeremystretch** [comment]
`primary_ip` is a read-only field that returns either the IPv4 or IPv6 address (depending on what's defined and the configured protocol preference).

To assign the primary IP(s) for a device or VM, you need to set `primary_ip4` and/or `primary_ip6` for IPv4 and IPv6 respectively.
